nupic/research/frameworks/vernon/distributed/experiments/supervised_experiment.py

```diff
@@ -0,0 +1,110 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import copy
+
+from torch.nn import DataParallel
+from torch.nn.parallel import DistributedDataParallel
+from torch.utils.data import DistributedSampler
+
+from nupic.research.frameworks.pytorch.distributed_sampler import (
+    UnpaddedDistributedSampler,
+)
+from nupic.research.frameworks.pytorch.model_utils import aggregate_eval_results
+from nupic.research.frameworks.vernon.distributed.experiments.components import (
+    DistributedBase,
+)
+from nupic.research.frameworks.vernon.experiments.supervised_experiment import (
+    SupervisedExperiment as SupervisedExperimentBase,
+)
+
+__all__ = [
+    "SupervisedExperiment",
+]
+
+
+class SupervisedExperiment(DistributedBase,
+                           SupervisedExperimentBase):
+    """
+    Distributed SupervisedExperiment.
```

Comment:
I was a bit confused here - the class name is SupervisedExperiment, but the docstring says Distributed SupervisedExperiment (judging from the inheritance it is the latter).

What does that mean, is Distributed the default way? How to run non-distributed Supervised Experiments?

Reply:
`vernon.experiments.SupervisedExperiment` is non-distributed. `vernon.distributed.experiments.SupervisedExperiment` is distributed.

Reply:
Got it, makes sense now, thanks.